src/input-device.c

```diff
@@ -0,0 +1,44 @@
+#include "input-device.h"
+
+#include <wayland-server.h>
+
+#include "zen-common.h"
+
+static void
+handle_device_destroy(struct wl_listener* listener, void* data)
```

Comment:
```suggestion
zn_input_device_handle_device_destroy(struct wl_listener* listener, void* data)
```

In our convention, if we consider this function to be a method of the `zn_input_device` object, we will start the function name with `zn_input_device`. If not, on the other hand, we believe that the variable name `self` should not be used.